src/node_file.cc

```diff
@@ -854,6 +872,43 @@ void FromNamespacedPath(std::string* path) {
 #endif
 }
 
+static inline int GetValidMode(Environment* env,
+                               Local<Value> mode_v,
+                               std::string_view type) {
+  if (!mode_v->IsInt32() && !mode_v->IsNullOrUndefined()) {
+    THROW_ERR_INVALID_ARG_TYPE(env, "mode must be int32 or null/undefined");
+    return -1;
+  }
+
+  int min = kMinimumAccessMode;
+  int max = kMaximumAccessMode;
+  int def = F_OK;
+
+  if (type == "copyFile") {
+    min = kMinimumCopyMode;
+    max = kMaximumCopyMode;
+    def = mode_v->IsNullOrUndefined() ? kDefaultCopyMode
+                                      : mode_v.As<Int32>()->Value();
+  } else if (type != "access") {
+    THROW_ERR_INVALID_ARG_TYPE(
+        env, "type must be equal to \"copyFile\" or \"access\"");
+    return -1;
+  }
+
+  if (mode_v->IsNullOrUndefined()) {
+    return def;
+  }
+
+  const int mode = mode_v.As<Int32>()->Value();
+  if (mode < min || mode > max) {
+    THROW_ERR_OUT_OF_RANGE(
+        env, "mode is out of range: >= %d && <= %d", min, max);
+    return -1;
+  }
+
+  return mode;
+}
```

Comment:
Consider using `v8::Maybe<int>` and returning `Just(result)` and `Nothing<int>()` (instead of result or -1) because that way callers can never mix up success and error cases.

Reply:
Agree with @bnoordhuis